tests/e2e/vmgroups_test.go

```diff
@@ -431,3 +432,143 @@ func (vg *VmGroupTest) TestVmGroupVolumeClone(c *C) {
+	}
+
+	out, err = adminutils.DeleteVMgroup(vg.config.EsxHost, vgTestVMgroup1, true)
+	c.Assert(err, IsNil, Commentf(out))
+
+	// vmgroup has been removed and all volumes belong to it have been removed
+	// "err.Error()" will be filled with "exit status 1"
+	volumeList := []string{vg.volName1, vg.volName2}
+	for _, volume := range volumeList {
+		out, err = verification.GetVMGroupForVolume(vg.config.EsxHost, volume)
+		log.Println("GetVMGroupForVolume return out[%s] err[%s] for volume %s", out, err, volume)
+		c.Assert(err.Error(), Equals, "exit status 1", Commentf("volume %s should be removed", volume))
+	}
+
+	// Recreate the test VM group1 and add VM1, VM2 back to the vmgroup
+	cmd := adminconst.CreateVMgroup + vgTestVMgroup1 + " --default-datastore " + vg.config.Datastores[0]
+	log.Printf("Creating test vmgroup %s", vgTestVMgroup1)
+	out, err = ssh.InvokeCommand(vg.config.EsxHost, cmd)
+	c.Assert(err, IsNil, Commentf(out))
+
+	for _, vm := range vmList {
+		out, err = adminutils.AddVMToVMgroup(vg.config.EsxHost, vgTestVMgroup1, vm)
+		c.Assert(err, IsNil, Commentf(out))
+	}
+	misc.LogTestEnd(c.TestName())
+}
+
+// test to verify volume is removed after user created vmgroup is removed without "--remove-volume" option
+// 1. create a user created vmgroup "vmgroup_test1" and add VMs to this vmgroup
+// 2. create two volumes for this vmgroup, one is on default datastore, and the other is on non-default datastore
+// 3. verify mount/unmount for this volume
+// 4. after unmount, verify the volume is detached for both volumes
+// 5. remove VMs from this vmgroup, and remove the vmgroup without "--remove-volume" option
+// 6. verify that volumes exist but do not belong to any vmgroup
+func (vg *VmGroupTest) TestVmgroupRemove(c *C) {
+	misc.LogTestStart(c.TestName())
+
+	// Create a volume in vgTestVMgroup1
+	vg.createVolumes(c, vg.volName1)
+
+	// Add create privilege for the second datastore
+	out, err := adminutils.AddCreateAccessForVMgroup(vg.config.EsxHost, vgTestVMgroup1, vg.config.Datastores[1])
+	c.Assert(err, IsNil, Commentf(out))
+
+	// Create another volume on second datastore
+	vg.createVolumes(c, vg.volName2+"@"+vg.config.Datastores[1])
+
+	// Verify volume can be mounted and unmounted for the first volume
+	out, err = dockercli.ExecContainer(vg.config.DockerHosts[0], vg.volName1, vg.testContainer)
+	c.Assert(err, IsNil, Commentf(out))
+
+	// Status should be detached
+	status := verification.VerifyDetachedStatus(vg.volName1, vg.config.DockerHosts[0], vg.config.EsxHost)
+	c.Assert(status, Equals, true, Commentf("Volume %s is not detached", vg.volName1))
+
+	// Verify volume can be mounted and unmounted for the second volume
+	out, err = dockercli.ExecContainer(vg.config.DockerHosts[0], vg.volName2+"@"+vg.config.Datastores[1], vg.testContainer)
+	c.Assert(err, IsNil, Commentf(out))
+
+	// Status should be detached
+	status = verification.VerifyDetachedStatusOnNonDefaultDS(vg.volName2, vg.config.Datastores[1],
```

Comment:
Same function as above can be used for the verification. Verification is best done by exec'ing a container on another docker host VM, which will be successful only when the volume is really detached from the VM.